Makefile

```diff
@@ -104,7 +104,7 @@ test: manifests generate fmt vet ## Run tests.
 
 build-within-docker: vendor
 	mkdir -p /build/cluster-api-provider-cloud-director
-	go build -ldflags "-X github.com/vmware/$(CAPVCD_IMG)/version.Version=${VERSION}" -o /build/vcloud/cluster-api-provider-cloud-director main.go
+	CGO_ENABLED=0 go build -ldflags "-s -w -X github.com/vmware/$(CAPVCD_IMG)/version.Version=${VERSION}" -o /build/vcloud/cluster-api-provider-cloud-director main.go
```

Comment:
Please consider _not_ strip debug symbols from the executable.

If you look at the community-maintained Cluster API providers, you will see that they do not strip debug symbols from their executables.

If the symbols are available, users are able to set breakpoints in the CAPVCD code, e.g., to understand how it works, or to investigate an issue.

If the symbols are not available, users have to either use their own CAPVCD images, or build CAPVCD and use debug symbols from a separate file, which, although possible, is complicated to do.